packages/core/src/modules/mdoc/MdocOptions.ts

```diff
@@ -0,0 +1,56 @@
+import type { Mdoc } from './Mdoc'
+import type { Key } from '../../crypto/Key'
+import type { DifPresentationExchangeDefinitionV2 } from '../dif-presentation-exchange'
+import type { ValidityInfo, MdocNameSpaces } from '@protokoll/mdoc-client'
+
+export type { MdocNameSpaces } from '@protokoll/mdoc-client'
+
+export type MdocVerifyOptions = {
+  trustedCertificates?: [string, ...string[]]
+  now?: Date
+}
+
+export type MdocOpenId4VpSessionTranscriptOptions = {
+  responseUri: string
+  clientId: string
+  verifierGeneratedNonce: string
+  mdocGeneratedNonce: string
+}
+
+export type MdocDeviceResponseOpenId4VpOptions = {
+  mdocs: [Mdoc, ...Mdoc[]]
+  presentationDefinition: DifPresentationExchangeDefinitionV2
+  deviceNameSpaces?: MdocNameSpaces
+  sessionTranscriptOptions: MdocOpenId4VpSessionTranscriptOptions
+}
+
+export type MdocDeviceResponseVerifyOptions = {
+  mdoc: Mdoc
+  trustedCertificates?: [string, ...string[]]
+  sessionTranscriptOptions: MdocOpenId4VpSessionTranscriptOptions
+  /**
+   * The base64Url-encoded device response string.
+   */
+  deviceResponse: string
+
+  /**
+   * The private part of the ephemeral key used in the session where the DeviceResponse was obtained. This is only required if the DeviceResponse is using the MAC method for device authentication.
```

Comment:
This is not the private part. It's a public key that can be used during verificadtion to know which private key to use.

```suggestion
   * The public ephemeral key used in the session where the DeviceResponse was obtained. This is only required if the DeviceResponse is using the MAC method for device authentication.
```

Also, isn't the public key added in the device respnose, or do we need to make sure it's MAC'd to this specific key?

Reply:
I removed this for now. As currently we never use mac authentication